applications/nrf_desktop/src/modules/profiler_sync.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <drivers/gpio.h>
+#include <profiler.h>
+
+#include "gpio_pins.h"
+
+#define MODULE profiler_sync
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_PROFILER_SYNC_LOG_LEVEL);
+
+#define SYNC_PERIOD_MIN		1000U
+#define SYNC_PERIOD_MAX		2000U
+#define SYNC_PERIOD_STEP	100U
+
+#define SYNC_EVENT_NAME		"sync_event"
+
+static struct device *gpio_dev;
+static struct gpio_callback gpio_cb;
+
+static struct k_delayed_work gen_sync_event;
+static u16_t sync_event_id;
+static bool pin_value;
+
+
+static void profile_sync_event(void)
+{
+	struct log_event_buf buf;
+
+	profiler_log_start(&buf);
+	profiler_log_send(&buf, sync_event_id);
+}
+
+static void register_sync_event(void)
+{
+	sync_event_id = profiler_register_event_type(SYNC_EVENT_NAME, NULL,
+						     NULL, 0);
+}
+
+static void gen_sync_event_isr(struct device *gpio_dev,
+			       struct gpio_callback *cb,
+			       u32_t pins)
+{
+	profile_sync_event();
+}
+
+static void gen_sync_event_fn(struct k_work *work)
+{
+	static u32_t sync_period = SYNC_PERIOD_MIN;
+
+	sync_period += SYNC_PERIOD_STEP;
+	if (sync_period > SYNC_PERIOD_MAX) {
+		sync_period = SYNC_PERIOD_MIN;
+	}
+
+	pin_value = !pin_value;
```

Comment:
if you use int you can always fo `pin_value = 1 - pin_value`

Reply:
done